did/util.go

```diff
@@ -62,12 +62,22 @@ func GetKeyFromVerificationMethod(did Document, kid string) (gocrypto.PublicKey,
 
 // matchesKIDConstruction checks if the targetID matches possible combinations of the did and kid
 func matchesKIDConstruction(did, kid, targetID string) bool {
-	maybeKID1 := kid                            // the kid == the kid
-	maybeKID2 := fmt.Sprintf("#%s", kid)        // the kid == the fragment with a #
-	maybeKID3 := fmt.Sprintf("%s#%s", did, kid) // the kid == the DID ID + the fragment with a #
-	maybeKID4 := fmt.Sprintf("%s%s", did, kid)  // the kid == the DID ID + the fragment without a #
-	maybeKID5, found := strings.CutPrefix(kid, did)
-	return targetID == maybeKID1 || targetID == maybeKID2 || targetID == maybeKID3 || targetID == maybeKID4 || (found && targetID == maybeKID5)
+	maybeKID1 := kid                                // the kid == the kid
```

Comment:
Cascading conditions make more sense now that there are 7 guesses. You can do that with a switch

Reply:
I tried that but found it much less readable...